src/libraries/System.Private.CoreLib/src/System/IO/FileStream.cs

```diff
@@ -107,27 +58,19 @@ public FileStream(IntPtr handle, FileAccess access, bool ownsHandle, int bufferS
                 throw;
             }
 
-            // Note: Cleaner to set the following fields in ValidateAndInitFromHandle,
-            // but we can't as they're readonly.
-            _access = access;
-            _useAsyncIO = isAsync;
-
-            // As the handle was passed in, we must set the handle field at the very end to
-            // avoid the finalizer closing the handle when we throw errors.
-            _fileHandle = safeHandle;
-        }
-
-        public FileStream(SafeFileHandle handle, FileAccess access)
-            : this(handle, access, DefaultBufferSize)
-        {
-        }
-
-        public FileStream(SafeFileHandle handle, FileAccess access, int bufferSize)
-            : this(handle, access, bufferSize, GetDefaultIsAsync(handle))
-        {
+            // it might seem to have no sense now, but we plan to introduce dedicated strategies for sync and async implementations
+            switch (isAsync)
+            {
+                case true:
+                    _impl = new FileStreamImpl(safeHandle, true, access, bufferSize, true);
```

Comment:
Please name the bool args, e.g. dontOwnHandle and isAsync.

ps dontOwnHandle is a strange name and polarity for that bool.  I suggest you invert its meaning and name it ownsHandle.